safenode/src/network/cmd.rs

```diff
@@ -74,43 +71,16 @@ impl SwarmDriver {
     pub(crate) fn handle_cmd(&mut self, cmd: SwarmCmd) -> Result<(), Error> {
         match cmd {
             SwarmCmd::GetProvidedData { key, sender } => {
-                match self.swarm.behaviour_mut().kademlia.store_mut().get(&key) {
-                    Some(data) => {
-                        sender
-                            .send(crate::protocol::messages::QueryResponse::GetChunk(Ok(
-                                Chunk::new(Bytes::from(data.value.clone())),
-                            )))
-                            .map_err(|_| Error::InternalMsgChannelDropped)?;
-                    }
-                    None => {
-                        // empty response here just to test out
-                        sender
-                            .send(crate::protocol::messages::QueryResponse::GetChunk(Err(
-                                ProtocolError::ProvideRecordNotFound,
-                            )))
-                            .map_err(|_| Error::InternalMsgChannelDropped)?;
-
-                        warn!("No data found in kademlia store for key: {:?}", key);
-                    }
-                }
+                let query_id = self.swarm.behaviour_mut().kademlia.get_record(key);
+                let _ = self.pending_query.insert(query_id, sender);
```

Comment:
So does this not actually the fire the request away into the network again here? 

I'm not sure we're actually getting data from _this node_?

Reply:
`get_record` will fire the query to network, and a response will be given back later.